client/lib/signup/flow-controller.js

```diff
@@ -130,6 +130,45 @@ assign( SignupFlowController.prototype, {
 		}.bind( this ) );
 	},
 
+	/**
+	 * This assertion helper validates that there are dependencies for the step provided
+	 * by completed steps
+	 * @param step The step to validate
+	 */
+	_assertStepDependencies( step ) {
```

Comment:
Take a look at that test:
https://github.com/Automattic/wp-calypso/pull/11401/files#diff-c530d5162d705f5bd3511598dbc47af2R108

In this scenario we're getting stuck and no error is thrown / displayed, because we haven't completed the flow, 
so we're not getting to the _assertFlowProvidedRequiredDependencies [check](https://github.com/Automattic/wp-calypso/blob/master/client/lib/signup/flow-controller.js#L168) because the `siteCreation` step is not completed and is waiting for the token that never supplied from the next step.